hadoop-hdds/common/src/main/java/org/apache/hadoop/utils/db/TypedTable.java

```diff
@@ -83,11 +116,22 @@ public boolean isEmpty() throws IOException {
 
   @Override
   public boolean isExist(KEY key) throws IOException {
-    CacheValue<VALUE> cacheValue= cache.get(new CacheKey<>(key));
-    return (cacheValue != null && cacheValue.getCacheValue() != null) ||
+
+    if (cacheCleanupPolicy == TableCacheImpl.CacheCleanupPolicy.NEVER) {
```

Comment:
I understand why we added this policy-specific check here. However it is probably misplaced here. Your original solution to have two cache types with `isExist` overloaded was probably better.